# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1319111672 -3600
#      Thu Oct 20 12:54:32 2011 +0100
# Node ID a2c74b5c02de68b9d8e48d56b117d70274be4dbf
# Parent  d13b6086aea9cc0f093e4f8fac3d2159cf5bb3fe
[MSVC] Prevent duplicate symbol.

* txt-eng-ft.cc: Include PermMatrix.h, only for MSVC.

diff --git a/src/txt-eng-ft.cc b/src/txt-eng-ft.cc
--- a/src/txt-eng-ft.cc
+++ b/src/txt-eng-ft.cc
@@ -50,16 +50,24 @@ gripe_missing_glyph (char c)
 static void
 gripe_glyph_render (char c)
 {
   warning_with_id ("Octave:glyph-render",
                    "ft_render: unable to render glyph for character `%c'",
                    c);
 }
 
+#ifdef _MSC_VER
+// This is just a trick to avoid multiply symbols definition.
+// PermMatrix.h contains a dllexport'ed Array<octave_idx_type>
+// that will make MSVC not to generate new instantiation and
+// use the imported one.
+#include "PermMatrix.h"
+#endif
+
 class
 ft_manager
 {
 public:
   static bool instance_ok (void)
     {
       bool retval = true;
 
